File: frontend/src/models/maintenance.model.ts
export interface AffectedEntity {
  entityType: string;
  entityName: string;
  entityID: number;
  closureStatus: boolean;
}

export interface Maintenance {
  [x: string]: any;
  maintenanceID: number;
  maintenanceStartDate: string;
  maintenanceEndDate?: string;
  reason: string;
  description: string;
  affectedEntities?: AffectedEntity[];
}

export interface MaintenanceEntity {
  entityID: number;
  entityName: string;
}

File: backend/Models/MaintenanceModel.cs
namespace backend.Models
{
  public class MaintenanceModel
  {
    public int MaintenanceID { get; set; }
    public DateTime MaintenanceStartDate { get; set; }
    public DateTime? MaintenanceEndDate { get; set; }
    public string? Reason { get; set; }
    public string Description { get; set; }
    public List<AffectedEntityModel>? AffectedEntities { get; set; }
  }

  public class AffectedEntityModel
  {
    public string EntityType { get; set; }
    public int EntityID { get; set; }
    public string EntityName { get; set; }
    public bool ClosureStatus { get; set; }
  }

}
